pulseapi/profiles/models/categories.py

```diff
@@ -59,3 +63,80 @@ class ProgramYear(models.Model):
 
     def __str__(self):
         return self.value
+
+
+class UserProfileProgramParticipation(models.Model):
+    profile = models.ForeignKey(
+        'profiles.UserProfile',
+        on_delete=models.CASCADE,
+        related_name='program_participation',
+    )
+
+    program = models.ForeignKey(
+        'profiles.ProgramType',
+        on_delete=models.CASCADE,
```

Comment:
I'm not sure CASCADE is the right policy here - cascading when the user gets deleted makes sense, but when a program gets deleted we probably want to preserve this "tuple" for historical "this user was in program X at some point, even though that program no longer exists"?

Reply:
I can change it. I was thinking that programs should never be deleted (even if it no longer exists).

Reply:
if the program has no associated users, deleting should be fine (because it doesn't alter the public-facing data in any way) but between "not deleting programs" and "cascading a delete", the former feels like the better policy.